### PR TITLE
Add swift static lib target to support AES GCM

### PR DESCRIPTION
**Pull request by rohitnarula7176**
## Proposed changes

Describe what this PR is trying to do.
Add swift static library target to support AES GCM.

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [x] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/IdentityCore.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1714,6 +1734,11 @@
...
 		1EE5413F2458B30300A86414 /* MSIDDevicePopManager.m */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.objc; path = MSIDDevicePopManager.m; sourceTree = "<group>"; };
+		1EE8FF5624F4BB3800CA1445 /* libIdentityCoreSwift.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; includeInIndex = 0; path = libIdentityCoreSwift.a; sourceTree = BUILT_PRODUCTS_DIR; };
+		1EE8FF5824F4BB3800CA1445 /* IdentityCoreSwift.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = IdentityCoreSwift.swift; sourceTree = "<group>"; };
+		1EE8FF6224F4C0E600CA1445 /* IdentityCoreTests iOS-Bridging-Header.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = "IdentityCoreTests iOS-Bridging-Header.h"; sourceTree = "<group>"; };
+		1EE8FF6324F4C0E600CA1445 /* IdentityCoreTests Mac-Bridging-Header.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = "IdentityCoreTests Mac-Bridging-Header.h"; sourceTree = "<group>"; };
+		1EE8FF6424F4C0E600CA1445 /* File.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = File.swift; sourceTree = "<group>"; };
```
Comment (oldalton) [inline IdentityCore/IdentityCore.xcodeproj/project.pbxproj:1741]: nit: can we call it something more representative other than File.Swift :)?

File: IdentityCore/IdentityCore.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -5711,7 +5782,35 @@
...
+			);
+			inputPaths = (
+			);
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "# Type a script or drag a script file from your workspace to insert its path.\n# target directory is Project root directory\ntarget_dir=${SRCROOT}/include/\n# Ensure the target include path exists\nmkdir -p ${target_dir}\n# Copy any file that looks like a Swift generated header to the include path\ncp ${DERIVED_SOURCES_DIR}/*-Swift.h ${target_dir}\n";
```
Comment (oldalton) [inline IdentityCore/IdentityCore.xcodeproj/project.pbxproj:5801]: Ok, so this filters out all Swift files and copies them to include folder? Any context on why this is needed?

File: IdentityCore/IdentityCoreSwift/IdentityCoreSwift.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+
+import Foundation
+/*
+the class file must be "open" so that it will be added to SwiftLibararyOne-Swift.h header file
+which will be used in Objective C file.
+*/
+@objc open class SwiftOneFileA: NSObject {
+@objc public func testOne() {
+      print("Swift One Library - File A - testOne function")
```
Comment (oldalton) [inline IdentityCore/IdentityCoreSwift/IdentityCoreSwift.swift:33]: can we remove this? looks like debug info.

File: IdentityCore/include/IdentityCoreSwift-Swift.h [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+#if __has_attribute(external_source_symbol)
+# pragma push_macro("any")
+# undef any
+# pragma clang attribute push(__attribute__((external_source_symbol(language="Swift", defined_in="IdentityCoreSwift",generated_declaration))), apply_to=any(function,enum,objc_interface,objc_category,objc_protocol))
+# pragma pop_macro("any")
+#endif
+
+
+SWIFT_CLASS("_TtC17IdentityCoreSwift13SwiftOneFileA")
+@interface SwiftOneFileA : NSObject
```
Comment (oldalton) [inline IdentityCore/include/IdentityCoreSwift-Swift.h:209]: Shouldn't this file be auto compiled? Should we update our gitginore to ignore it?

File: IdentityCore/IdentityCoreSwift/IdentityCoreSwift.swift (hunk 3 above)
Comment (rohitnarula7176) [reply IdentityCore/IdentityCoreSwift/IdentityCoreSwift.swift:33]: Yes I added a dummy method to show Kevin how he can use this class in the tests.

File: IdentityCore/IdentityCore.xcodeproj/project.pbxproj (hunk 2 above)
Comment (rohitnarula7176) [reply IdentityCore/IdentityCore.xcodeproj/project.pbxproj:5801]: It is how xcode creates a Objective header for the swift file and adds it to the derived sources folder under the IdentityCoreSwift target. We need a script to move that file to the SOURCE_DIR so the IdentityCoreTests can reference it in the header search paths and use it accordingly.

Comment (rohitnarula7176) [reply IdentityCore/IdentityCore.xcodeproj/project.pbxproj:5801]: Please refer to this .https://medium.com/@mail2ashislaha/swift-objective-c-interoperability-static-libraries-modulemap-etc-39caa77ce1fc

File: IdentityCore/IdentityCore.xcodeproj/project.pbxproj (hunk 1 above)
Comment (rohitnarula7176) [reply IdentityCore/IdentityCore.xcodeproj/project.pbxproj:1741]: Yes this was a dummy file needed like we discussed to generate bridging header. I can maybe rename it to Dummy.swift. :)

File: IdentityCore/include/IdentityCoreSwift-Swift.h (hunk 4 above)
Comment (rohitnarula7176) [reply IdentityCore/include/IdentityCoreSwift-Swift.h:209]: Yes we can move this to giitignore. I will add it to gitignore.

File: IdentityCore/IdentityCore.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -6949,6 +7055,133 @@
...
+				SWIFT_ACTIVE_COMPILATION_CONDITIONS = DEBUG;
+				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
+				SWIFT_VERSION = 5.0;
+				TARGETED_DEVICE_FAMILY = "1,2";
+			};
+			name = Debug;
+		};
+		1EE8FF5B24F4BB3800CA1445 /* Release */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
```
Comment (oldalton) [inline IdentityCore/IdentityCore.xcodeproj/project.pbxproj:7127]: is this target also using xcconfig?

File: IdentityCore/IdentityCoreSwift/IdentityCoreSwift.swift (hunk 3 above)
Comment (oldalton) [reply IdentityCore/IdentityCoreSwift/IdentityCoreSwift.swift:33]: ok, let's remove this one.

File: IdentityCore/include/IdentityCoreSwift-Swift.h (hunk 4 above)
Comment (oldalton) [reply IdentityCore/include/IdentityCoreSwift-Swift.h:209]: let's remove the file too

File: IdentityCore/IdentityCore.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -6949,6 +7055,133 @@
...
+				CLANG_WARN_CONSTANT_CONVERSION = YES;
+				CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES;
+				CLANG_WARN_DIRECT_OBJC_ISA_USAGE = YES_ERROR;
+				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
+				CLANG_WARN_EMPTY_BODY = YES;
+				CLANG_WARN_ENUM_CONVERSION = YES;
+				CLANG_WARN_INFINITE_RECURSION = YES;
+				CLANG_WARN_INT_CONVERSION = YES;
+				CLANG_WARN_NON_LITERAL_NULL_CONVERSION = YES;
+				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
```
Comment (NerevarineRule) [inline IdentityCore/IdentityCore.xcodeproj/project.pbxproj:7148]: this setting is here too; interesting.

File: IdentityCore/include/IdentityCoreSwift-Swift.h [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+# define SWIFT_TYPEDEFS 1
+# if __has_include(<uchar.h>)
+#  include <uchar.h>
+# elif !defined(__cplusplus)
+typedef uint_least16_t char16_t;
+typedef uint_least32_t char32_t;
+# endif
+typedef float swift_float2  __attribute__((__ext_vector_type__(2)));
+typedef float swift_float3  __attribute__((__ext_vector_type__(3)));
+typedef float swift_float4  __attribute__((__ext_vector_type__(4)));
```
Comment (NerevarineRule) [inline IdentityCore/include/IdentityCoreSwift-Swift.h:38]: wow fancy stuff vector literals.

Comment (rohitnarula7176) [reply IdentityCore/include/IdentityCoreSwift-Swift.h]: This is an auto generated file.

File: IdentityCore/IdentityCore.xcodeproj/project.pbxproj (hunk 6 above)
Comment (rohitnarula7176) [reply IdentityCore/IdentityCore.xcodeproj/project.pbxproj:7148]: Yes it gets added by default.

Comment (rohitnarula7176) [reply IdentityCore/IdentityCore.xcodeproj/project.pbxproj:7148]: Also the file used to enable AES GCM will be IdentityCoreSwift.swift under IdentityCore target, Kevin from cpp team already has the code ready and he will just add that code to this file.

**Pull request merged**